lib/board.rb

```diff
@@ -0,0 +1,62 @@
+class Board
+  attr_accessor :cells
+
+
+
+  def initialize
+    self.reset!
+
+  end
+
+  def reset!
+    @cells = [" "," "," "," "," "," "," "," "," "]
+  end
+
+  def display
+    b = @cells
+    puts " #{b[0]} | #{b[1]} | #{b[2]} "
+    puts "-----------"
+    puts " #{b[3]} | #{b[4]} | #{b[5]} "
+    puts "-----------"
+    puts " #{b[6]} | #{b[7]} | #{b[8]} "
+  end
+
+  def position(pos)
+    @cells[pos.to_i-1]
+  end
+
+  def full?
+    @cells.all? do |pos|
+      pos == "X" || pos == "O"
+    end
+  end
+
+  def turn_count
+    count = 0
+    @cells.each do |pos|
+      count +=1 if (pos != " ")
```

Comment:
Consider using the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |pos| #logic here }
end
```